### PR TITLE
Add volume backup result

### LINKED ISSUE #7676: Add volume result in backup's VolumeInfo
**Describe the problem/challenge you have**

The current backup's VolumeInfo doesn't reflect the Volume data backup result. It's better to display that too.

**Describe the solution you'd like**

**Anything else you would like to add:**

**Environment:**

- Velero version (use `velero version`):
- Kubernetes version (use `kubectl version`):
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "The project would be better with this feature added"
- :-1: for "This feature will not enhance the project in a meaningful way"

### PR DESCRIPTION
**Pull request by blackpiglet**
Thank you for contributing to Velero!

# Please add a summary of your change
* Add the result in the backup's VolumeInfo.
* Add more prow actions.

# Does your change fix a particular issue?

Fixes #7676 

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Comment by reasonerjt** [comment]
Let's revisit to determine if it should target v1.14.1

File: internal/volume/volumes_information.go [hunk 1]
```diff
@@ -92,6 +92,9 @@ type BackupVolumeInfo struct {
 	// Snapshot completes timestamp.
 	CompletionTimestamp *metav1.Time `json:"completionTimestamp,omitempty"`
 
+	// Whether the volume data is backed up successfully.
+	Succeeded bool `json:"succeeded,omitempty"`
```
Comment (reasonerjt) [inline internal/volume/volumes_information.go:96]: Instead of trying to interpret the result we may leave it as is it the specific "XXXInfo" struct.

Comment (Lyndon-Li) [inline internal/volume/volumes_information.go:96]: For long term consideration, there are at lease 3 terminating states --- succeeded, failed, cancelled. So could we consider to make it as a string?

File: .github/workflows/prow-action.yml
Comment (kaovilai) [inline .github/workflows/prow-action.yml:1]: Do you mean to change this file here?

Comment (blackpiglet) [reply .github/workflows/prow-action.yml:1]: This PR was created for a while.
There are some out-of-date changes.
Thanks. I deleted this file.

File: changelogs/unreleased/7775-blackpiglet [hunk 2]
```diff
@@ -0,0 +1 @@
+Add the result in the backup's VolumeInfo. Add more prow actions.
```
Comment (kaovilai) [inline changelogs/unreleased/7775-blackpiglet:1]:
```suggestion
Add the result in the backup's VolumeInfo.
```

File: internal/volume/volumes_information.go [hunk 3]
```diff
@@ -92,13 +93,24 @@ type BackupVolumeInfo struct {
...
 	PVBInfo                  *PodVolumeInfo            `json:"pvbInfo,omitempty"`
 	PVInfo                   *PVInfo                   `json:"pvInfo,omitempty"`
 }
 
+type VolumeResult string
+
+const (
+	VolumeResultSucceeded VolumeResult = "succeeded"
+	VolumeResultFailed    VolumeResult = "failed"
+	VolumeResultCancelled VolumeResult = "canceled"
```
Comment (kaovilai) [inline internal/volume/volumes_information.go:111]:
```suggestion
	VolumeResultCanceled VolumeResult = "canceled"
```
matching spelling of canceled on either side of `=`

File: changelogs/unreleased/7775-blackpiglet (hunk 2 above)
Comment (blackpiglet) [reply changelogs/unreleased/7775-blackpiglet:1]: Thanks. Modified.

File: internal/volume/volumes_information.go (hunk 1 above)
Comment (blackpiglet) [reply internal/volume/volumes_information.go]: Modified.

File: internal/volume/volumes_information.go (hunk 3 above)
Comment (blackpiglet) [reply internal/volume/volumes_information.go]: Modified.

File: internal/volume/volumes_information.go (hunk 1 above)
Comment (blackpiglet) [reply internal/volume/volumes_information.go]: Add the original phase in the `XXXInfo` struct and keep the `Result` parameter.

**Review by reasonerjt — APPROVED** [review]

**Review by Lyndon-Li — APPROVED** [review]

File: internal/volume/volumes_information.go [hunk 4]
```diff
@@ -92,13 +92,24 @@ type BackupVolumeInfo struct {
...
 	PVBInfo                  *PodVolumeInfo            `json:"pvbInfo,omitempty"`
 	PVInfo                   *PVInfo                   `json:"pvInfo,omitempty"`
 }
 
+type VolumeResult string
+
+const (
+	VolumeResultSucceeded VolumeResult = "succeeded"
+	VolumeResultFailed    VolumeResult = "failed"
+	//VolumeResultCanceled  VolumeResult = "canceled"
```
Comment (kaovilai) [inline internal/volume/volumes_information.go:110]: Do we have plans to add cancel in the future?

Comment (blackpiglet) [reply internal/volume/volumes_information.go:110]: It's possible, but there is no plan yet.

**Pull request merged**